### PR TITLE
Add tool to detect missing files.

### PR DESCRIPTION
**Pull request by fightingtexasaggie**

### CODE REVIEW

File: src/Command/PelagosCheckForMissingFilesCommand.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        foreach ($datasets as $dataset) {
+            $checkedUdi = $dataset->getUdi();
+            $io->note("checking dataset: $checkedUdi\n");
+            /** @var Dataset $dataset */
+            $datasetSubmission = $dataset->getDatasetSubmission();
+            if ($datasetSubmission instanceof DatasetSubmission) {
+                $fileset = $datasetSubmission->getFileset();
+                if ($fileset instanceof Fileset) {
+                    $files = $fileset->getAllFiles()->filter(function (File $file) {
+                        return $file->getStatus() !== File::FILE_ERROR;
```
Comment (mickel1138) [inline src/Command/PelagosCheckForMissingFilesCommand.php:94]: How about new files, or file in queue?

File: src/Command/PelagosCheckForMissingFilesCommand.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                    });
+                    $fileCount = count($files);
+                    $io->writeln('Dataset ' . $dataset->getUdi() . ' has ' . $fileCount . ' files.' . "\n");
+                    /** @var File $file */
+                    foreach ($files as $file) {
+                        $originalFile = $file->getPhysicalFilePath();
+                        if (substr($originalFile, 0, 1) != '/') {
+                            $originalFile = $this->dataStoreDirectory . "/$originalFile";
+                        }
```
Comment (mickel1138) [inline src/Command/PelagosCheckForMissingFilesCommand.php:103]: Let the data store check the existence, don't do it like this. `filesystem->has`

File: src/Command/PelagosCheckForMissingFilesCommand.php (hunk 1 above)
Comment (fightingtexasaggie) [reply src/Command/PelagosCheckForMissingFilesCommand.php:94]: This should be getting them...any file not in error.

File: src/Command/PelagosCheckForMissingFilesCommand.php (hunk 2 above)
Comment (fightingtexasaggie) [reply src/Command/PelagosCheckForMissingFilesCommand.php:112]: switching to Filesystem component for this.

File: src/Command/PelagosCheckForMissingFilesCommand.php (hunk 1 above)
Comment (fightingtexasaggie) [reply src/Command/PelagosCheckForMissingFilesCommand.php]: In the end, decided to make it just check the done files.

File: config/services.yaml [hunk 3]
```diff
@@ -125,6 +125,11 @@ services:
             string $passwordExpiryWarn: "%env(ACCOUNT_PASSWORD_EXPIRATION_WARN)%"
             string $hostName: "%env(HOSTNAME)%"
 
+    App\Command\PelagosCheckForMissingFilesCommand:
+        bind:
+            string $dataStoreDirectory: "%env(DATA_STORE_DIRECTORY)%"
+            $datastoreFlysystem: "@oneup_flysystem.datastore_flysystem_filesystem"
```
Comment (praneethpr) [inline config/services.yaml:131]: We should inject src/Util/Datastore, we shouldn't be using it directly either from a controller or command

File: src/Command/PelagosCheckForMissingFilesCommand.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                if ($fileset instanceof Fileset) {
+                    $files = $fileset->getAllFiles()->filter(function (File $file) {
+                        return $file->getStatus() === File::FILE_DONE;
+                    });
+                    $fileCount = count($files);
+                    $io->writeln('Dataset ' . $dataset->getUdi() . ' has ' . $fileCount . ' files.' . "\n");
+                    /** @var File $file */
+                    foreach ($files as $file) {
+                        $filePath = $file->getPhysicalFilePath();
+                        if (!$this->datastoreFlysystem->has($filePath)) {
```
Comment (fightingtexasaggie) [inline src/Command/PelagosCheckForMissingFilesCommand.php:112]: Extend the Datastore util instead!

**Review by praneethpr — DISMISSED** [review]

**Review by mickel1138 — APPROVED** [review]

**Pull request merged**